src/main/java/shef/servlets/NewLiveStreamServlet.java

```diff
@@ -25,19 +25,21 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet responsible for creating new livestreams. */
+/** Servlet responsible for creating new LiveStream entities
+    and storing them in Datastore. */
 @WebServlet("/new-live-stream")
 public class NewLiveStreamServlet extends HttpServlet {
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String recipeKey = request.getParameter("recipe-selection");
-    String link = request.getParameter("live-stream-link");
-
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Entity liveStreamEntity = new Entity("LiveStream");
-    liveStreamEntity.setProperty("recipe-key", recipeKey);
-    liveStreamEntity.setProperty("link", link);
-
+    if (request.getParameterMap().get("duration")[0].equals("P0D")) {
+      throw new IllegalArgumentException("Live stream start or end time is missing or invalid.");
+    }
+    for (String param : request.getParameterMap().keySet()) {
```

Comment:
Since you're accessing the parameter map multiple times here, I'd recommend saving it into a variable so you can reference the same map in each iteration of the loop instead of creating a new and identical one each time

Reply:
Fixed